transport/tchannel/peer.go

```diff
@@ -171,10 +173,15 @@ func (p *tchannelPeer) getPeer() *tchannel.Peer {
 }
 
 func (p *tchannelPeer) getRootPeers() *tchannel.RootPeerList {
-	if p.transport.ch == nil {
+	ch := p.ch
+	if ch == nil {
```

Comment:
maybe a nested loop will be cleaner but your call. On first pass, I was confused why we check `ch` twice
```go
if ch == nil {
    if p.transport.ch == nil {
        return nil
    }
    ch = p.transport.ch
}
```